custom_components/octopus_energy/heat_pump/zone.py

```diff
@@ -89,42 +92,49 @@ def _handle_coordinator_update(self) -> None:
     """Retrieve the previous rate."""
 
     current = now()
-    result: HeatPumpCoordinatorResult = self.coordinator.data if self.coordinator is not None and self.coordinator.data is not None else None
+    result: HeatPumpCoordinatorResult = self.coordinator.data if self.coordinator is not None and self.coordinator.data is not None else None    
     if (result is not None and 
         result.data is not None and 
-        result.data.octoHeatPumpControllerConfiguration is not None and
-        result.data.octoHeatPumpControllerConfiguration.zones and
+        result.data.octoHeatPumpControllerStatus is not None and
```

Comment:
question: Was there a reason this was changed? 

From my understanding from the data, the configuration current operature was equal to the status telemetry. I originally had it like this, but when I needed to capture the `currentOperature.end` for boosting purposes, I changed it to the current way so that I wouldn't have to do two loops (reduced efficiency, arguably negligible) but also incase one was available and the other wasn't which could cause inconsistencies.

If there wasn't a reason, I think it should be changed back to use the `octoHeatPumpControllerConfiguration`

Reply:
The relay status isn't available from `octoHeatPumpControllerConfiguration`, and I chose to change for the same consistency reasons given the majority of the values here are in/can be retrieved from the status API. It's a bit strange that the same telemetry data exists in both.

Reply:
Gotta love an inconsistent API :(